Carrot-Assistant/components/models.py

```diff
@@ -49,39 +139,565 @@ def get_model(
     object
         An interface to generate text using an LLM
     """
-    if "gpt" in model_name.lower():
-        logger.info(f"Loading {model_name} model")
+    # Normalize the huggingface for downloading the model
+    from huggingface_hub import hf_hub_download
 
-        llm = OpenAIGenerator(
-            model=model_name, generation_kwargs={"temperature": temperature}
-        )
+    # Normalize the model name to lowercase
+    model_name = model_name.lower()
 
-    else:
-        logger.info(f"Loading {model_name} model")
-        from huggingface_hub import hf_hub_download
+    # Select device based on GPU availability
+    device = -1 if torch.cuda.is_available() else 0
 
-        device = -1 if torch.cuda.is_available() else 0
+    try:
+        # ------ GPT Model ------ >
+        # --------------------------
 
-        try:
-            llm = LlamaCppGenerator(
-                model=hf_hub_download(**local_models[model_name]),
```

Comment:
Just put it back to this and it should still work

Reply:
Sure. I'll work on it.